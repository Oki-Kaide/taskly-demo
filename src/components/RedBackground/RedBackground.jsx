import React from 'react';
import PropTypes from 'prop-types';
import './RedBackground.sass';

const RedBackground = ({side, children, top, bottom, right, left}) => {
  return (
    <div className={`red-image ${side}`}>
      <div className={`red-image-content ${side}`} style={{ top, bottom, right, left }}>
        {children}
      </div>
    </div>
  );
}

RedBackground.propTypes = {
  side: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  top: PropTypes.string,
  bottom: PropTypes.string,
  right: PropTypes.string,
  left: PropTypes.string,
}

RedBackground.defaultProps = {
  top: 'auto',
  bottom: 'auto',
  right: 'auto',
  left: 'auto',
}

export default RedBackground;