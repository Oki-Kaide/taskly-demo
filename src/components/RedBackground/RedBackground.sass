@import '../../styles/variables.sass'
@import '../../styles/mixins.sass'

.red-image
  position: fixed
  top: 0
  height: 100vh
  background-color: $taskly-red
  content: ''
  @include for-phone-only
    display: none
  @include for-tablet-portrait-up
    display: none
  @include for-tablet-portrait-landscape-up
    width: 40%
    display: unset
  @include for-tablet-landscape-up
    width: calc(100vw / 2)
    display: unset


  &.right
    border-top-left-radius: 100px
    border-bottom-left-radius: 100px
    right: 0

  &.left
    border-top-right-radius: 100px
    border-bottom-right-radius: 100px
    left: 0
  
  .red-image-content
    position: fixed
    img
      width: calc(100vw / 2)
