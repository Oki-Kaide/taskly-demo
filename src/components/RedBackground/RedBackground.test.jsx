import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RedBackground from './RedBackground';

const createTestProps = props => ({
  side: 'left',
  children: <div />,
  top: '10px',
  bottom: '10px',
  right: '50px',
  left: '25px',
  ...props,
});

describe('RedBackground component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<RedBackground {...props} />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<RedBackground {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

});
