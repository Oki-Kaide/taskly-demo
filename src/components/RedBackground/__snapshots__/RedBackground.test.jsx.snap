// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedBackground component matches the rendered snapshot 1`] = `
<div
  className="red-image left"
>
  <div
    className="red-image-content left"
    style={
      Object {
        "bottom": "10px",
        "left": "25px",
        "right": "50px",
        "top": "10px",
      }
    }
  >
    <div />
  </div>
</div>
`;
